src/narrowdown.h

```diff
@@ -81,9 +81,9 @@ class NarrowDown
 
   static std::string shortestStringInBetween(const std::string& a, const std::string& b)
   {
-    ASSERT(a, <, b);
+    ASSERT(a, <=, b);
     const auto m = std::mismatch(a.begin(), a.end(), b.begin());
-    return std::string(b.begin(), m.second+1);
+    return std::string(b.begin(), std::min(b.end(), m.second+1));
```

Comment:
This comment is about the whole commit, not the code but I must comment on code.

I'm not sure we need this (or that we should handle it this way).
URl's items must be unique. If not, this is a error in the zim file.
Older implementations of libzim creator were not enforce this but "newer" one does.
So no zimfile should come in this situation. Only old ones may and we should consider them as broken.

I didn't know about a zim file with this problem. Now we have one and it is a "fake" one.
We should probably raise a ZimFormatError here.

Reply:
I don't have any preference with regard to that detail. Please feel free to change the code as you like.

Reply:
Let's keep it. Even if the zim file is not theoretically valid, it will allow the user to read most of the entries.